.github/workflows/release-trigger.yaml

```diff
@@ -0,0 +1,31 @@
+name: Check Version Update
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  check-version:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Push tag for each updated package
+        run: |
+          git config --global user.name "GitHub Action"
+          git config --global user.email "github-action@github.com"
+
+          for file in $(git diff --name-only HEAD~1..HEAD | grep pyproject.toml); do    
+              if git diff HEAD~1..HEAD $file | grep "version ="; then
```

Comment:
Can't this also be triggered by formatting changes e.g.? What will happen if the tag already exists?

Reply:
You are correct formatting changes are a problem. I changed it so it now checks if the version actually has changed.
If the tag already exists the pipeline will fail. I think that's ok, if that happens something went wrong and we don't want to overwrite tags